### PR TITLE
Add macos arm64 runner

### PR DESCRIPTION
**Pull request by cd-work**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@ on:
   push:
     branches:
       - main
+      - eternal_ci
```
Comment (cd-work) [inline .github/workflows/release.yml:12]: This is solely for the purpose of demonstrating this PR works.

Comment (kylewillmon) [reply .github/workflows/release.yml:12]: That's what we've got the `workflow_dispatch` trigger for. You can always manually run this workflow on any branch [from the actions tab](https://github.com/phylum-dev/cli/actions/workflows/release.yml)

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -46,14 +47,14 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          # - target: x86_64-unknown-linux-musl
-          #   os: ubuntu-latest
+          - target: x86_64-unknown-linux-musl
+            os: ubuntu-latest
           # - target: aarch64-unknown-linux-musl
           #   os: ubuntu-latest
```
Comment (kylewillmon) [inline .github/workflows/release.yml:53]: Any reason that we can't re-enable this one on a self-hosted runner as well?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -46,14 +47,14 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          # - target: x86_64-unknown-linux-musl
-          #   os: ubuntu-latest
+          - target: x86_64-unknown-linux-musl
+            os: ubuntu-latest
```
Comment (kylewillmon) [inline .github/workflows/release.yml:51]: This needs to be `-gnu` now, right?

Comment (cd-work) [reply .github/workflows/release.yml:51]: I actually didn't want to comment this out, I'll fix all that up in @andreaphylum's PR.

File: .github/workflows/release.yml (hunk 2 above)
Comment (kylewillmon) [reply .github/workflows/release.yml:53]: Answered in other comment

**Pull request merged**